App/Logic/AnalyticsLogic.swift

```diff
@@ -199,11 +300,76 @@ extension Logic.Analytics {
   }
 
   /// Updates the opportunity window for the event without exposure
-  struct UpdateEventWithoutExposureOppurtunityWindow: AppStateUpdater {
+  struct SetEventWithoutExposureOpportunityWindow: AppStateUpdater {
     let window: AnalyticsState.OpportunityWindow
 
     func updateState(_ state: inout AppState) {
       state.analytics.eventWithoutExposureWindow = self.window
     }
   }
+
+  /// Sets the opportunity window for the dummy analytics traffic
+  struct SetDummyTrafficOpportunityWindow: AppStateUpdater {
+    let dummyTrafficStochasticDelay: Double
+    let now: Date
+
+    func updateState(_ state: inout AppState) {
+      let windowStart = self.now.addingTimeInterval(self.dummyTrafficStochasticDelay)
+      let windowDuration = AnalyticsState.OpportunityWindow.secondsInDay
+      state.analytics.dummyTrafficOpportunityWindow = .init(windowStart: windowStart, windowDuration: windowDuration)
+    }
+  }
+}
+
+// MARK: Helpers
+
+private extension AnalyticsRequest.Body {
+  /// Creates a dummy request
+  static func dummy(deviceTokenLength: Int) -> Self {
+    return Self(
+      province: Province.allCases.randomElement() ?? AppLogger.fatalError("No provinces defined"),
+      exposureNotificationStatus: ExposureNotificationStatus.randomCase(),
+      pushNotificationStatus: UNAuthorizationStatus.randomCase(),
+      riskyExposureDetected: Bool.random(),
+      deviceToken: String.random(length: deviceTokenLength)
+    )
+  }
+}
+
+private extension ExposureNotificationStatus {
+  static let allCases: [Self] = [
```

Comment:
Uhm, why not using `CaseIterable`?

Reply:
You should be able to remove all these extensions and use `Enum.allCases.randomElement()`

Reply:
Given that I can't privately conform to a protocol, that would mean making exposure notification status CaseIterable where it's defined, which I feel would be misleading. There's a comment right below.

If you feel like this shouldn't be the case I'm fine with changing it

Reply:
I don't see the problem honestly in conforming to CaseIterable

Reply:
Done. I had to wrap `UNAuthorizationStatus` though